library/types/test/hostname_test.rb

```diff
@@ -10,30 +10,49 @@
 
   let(:etc_hostname) { "etc.hostname.cz" }
   let(:cmd_hostname) { "cmd.hostname.cz" }
+  let(:hostname)     { Yast::Hostname }
+
+  def allow_execute_hostname(result)
+    ret = { "stdout" => result, "exit" => 0 } if result
 
-  it "returns output of hostname --fqdn if available" do
     allow(Yast::SCR)
       .to receive(:Execute)
       .with(path(".target.bash_output"), "hostname --fqdn")
-      .and_return("stdout" => cmd_hostname, "exit" => 0)
-
-    expect(Yast::Hostname.CurrentFQ).to eq cmd_hostname
+      .and_return(ret)
```

Comment:
so it returns nil if result if nil? I think it is not correct behavior of SCR. I think simple adding `result = ""` helps with issue ( and remove such if result ).
better solution for me if you want also test error states is
```
def allow_execute_hostname(stdout: "", exit: 0, stderr: "")
end
```

so you have reasonable defaults and can be in test easily changed.